src/planscape/planning/views.py

```diff
@@ -164,38 +196,51 @@ def delete_planning_area(request: HttpRequest) -> HttpResponse:
         # Check that the user is logged in.
         user = request.user
         if not user.is_authenticated:
-            return JsonResponse({"error": "Authentication Required"}, status=401)
+            return Response(
+                {"error": "Authentication Required"},
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
 
         # Get the planning area IDs
         body = json.loads(request.body)
         planning_area_id = body.get("id", None)
         planning_area_ids = []
         if planning_area_id is None:
-            raise ValueError("Must specify planning area id.")
+            return Response(
+                {"message": "No ID given for planning area"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
         elif isinstance(planning_area_id, int):
             planning_area_ids = [planning_area_id]
         elif isinstance(planning_area_id, list):
             planning_area_ids = planning_area_id
         else:
-            raise ValueError("Planning Area ID must be an int or a list of ints.")
-
-        # Get the planning area(s) for just the logged in user.
-        planning_areas = user.planning_areas.filter(pk__in=planning_area_ids)
+            return Response(
+                {"message": "Planning Area ID must be an int or a list of ints."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
 
-        planning_areas.delete()
+        planning_areas = PlanningArea.objects.get_for_user(user).filter(
+            pk__in=planning_area_ids
+        )
+        for p in planning_areas:
+            if PlanningAreaPermission.can_remove(user, p):
+                p.delete()
+            else:
+                logger.error(f"User {user} has no permission to delete {p.id}")
 
         # We still report that the full set of planning area IDs requested were deleted,
         # since from the user's perspective, there are no planning areas with that ID.
         # The end result is that those planning areas don't exist as far as the user is concerned.
-        response_data = {"id": planning_area_ids}
+        return Response({"id": planning_area_ids}, content_type="application/json")
 
-        return HttpResponse(json.dumps(response_data), content_type="application/json")
     except Exception as e:
-        return HttpResponseBadRequest("Error in delete: " + str(e))
+        logger.exception(f"Error deleting planningarea: {e}")
+        raise
```

Comment:
I'm also adopting this pattern where uncaught situations re-raise the Exception. I can also swallow this and call sentry directly if we just want to flag an issue gracefully.